src/core/globalstate.ts

```diff
@@ -45,6 +45,12 @@ export class MobXGlobals {
 	isRunningReactions = false;
 
 	/**
+	 * Current reaction (if any is running).
+	 * Reactions are run after derivations using a trampoline.
+	 */
+	currentReaction: Reaction = undefined;
```

Comment:
bq. Would a viable alternative to this problem be always passing in the reaction object to the callback functions of reactions?
I think, that's even better then my initial proposition! autorun can work the same way:
```
autorun(r => {   r.dispose();  })
```